addons/web/static/tests/fields/relational_fields/field_one2many_tests.js

```diff
@@ -9595,24 +9596,22 @@ QUnit.module('fields', {}, function () {
             const prom = new Promise(r => {
                 resolveCP = r;
             });
-            ControlPanel.patch('cp_patch_mock', T =>
-                class extends T {
-                    constructor() {
-                        super(...arguments);
-                        owl.hooks.onMounted(() => {
-                            assert.step('mounted');
-                        });
-                        owl.hooks.onWillUnmount(() => {
-                            assert.step('willUnmount');
-                        });
-                    }
-                    async update() {
-                        // the issue is a race condition, so we manually delay the update to turn it deterministic
-                        await prom;
-                        super.update(...arguments);
-                    }
-                }
-            );
+            const unpatch = patch(ControlPanel, 'cp_patch_mock', {
+                setup() {
```

Comment:
The constructor from the old code is replaced by this `setup` method. But I can't find any call to this method. I'm expecting that it is called in the `constructor` of the patched class, but the patched class's constructor isn't modified to call `setup`. Do patched classes automatically call `setup` during instantiation?

(Sorry to just come out of the blue. I'm just very excited to this newly improved feature of patching. :) )

Reply:
it's now always done for component by Owl itself: https://github.com/odoo/owl/commit/19a47a7001a90ca6fd6a1a668b7107e5af4c74bf (you need the last version of owl for that).  With this, all components can be patched easily.  In odoo, other classes can define a `setup` method if necessary.

Reply:
I see. Thanks.